contributed_definitions/nyaml/NXmpes.yaml

```diff
@@ -2,62 +2,150 @@ category: application
-        doc: |
-          Author ID defined by https://orcid.org/.
-    (NXinstrument):
-      energy_resolution(NX_FLOAT):
-        unit: NX_ENERGY
-      (NXsource):
+        vendor:
+          exists: recommended
+        model:
+          exists: recommended
+        identifier:
+          exists: recommended
+      source_TYPE(NXsource):
+        
+        # Comment from mpes may workshop:
+        # - There is much more information which can be provided for NXsource
         doc: |
           The source used to generate the primary photons. Properties refer strictly to
           parameters of the source, not of the output beam. For example, the energy of the
           source is not the optical power of the beam, but the energy of the electron beam
           in a synchrotron and so on.
         type:
-          enumeration: [Synchrotron X-ray Source, Rotating Anode X-ray, Fixed Tube X-ray, UV Laser, Free-Electron Laser, Optical Laser, UV Plasma Source, Metal Jet X-ray, HHG laser]
+          enumeration: [Synchrotron X-ray Source, Rotating Anode X-ray, Fixed Tube X-ray, UV Laser, Free-Electron Laser, Optical Laser, UV Plasma Source, Metal Jet X-ray, HHG laser, UV lamp, Monochromatized electron source, other]
+        type_other:
+          exists: optional
+          doc: |
+            Specification of type, may also go to name.
         name:
+          exists: recommended
         probe:
+          exists: optional
           doc: |
             Type of probe. In photoemission it's always photons, so the full NIAC list is
             restricted.
-          enumeration: [x-ray, ultraviolet, visible light]
-      (NXbeam):
+          enumeration: [photon, x-ray, visible light, ultraviolet]
+        device_information(NXfabrication):
+          exists: recommended
+          vendor:
+            exists: recommended
+          model:
+            exists: recommended
+          identifier:
+            exists: recommended
+        associated_beam(NXbeam):
+          doc: |
+            The beam emitted by this source.
+            Should be named with the same appendix, e.g.,
+            for `source_probe` it should refer to `beam_probe`.
+            Refers to the same concept as /NXentry/NXinstrument/beam_TYPE
+            and may be linked.
+      
+      # It would be nice if we had the notion of linking in nexus or if
+      # we have a subdefinition NXlink to refer to a path.
+      # This would also be helpful for NXtransformations
+      beam_TYPE(NXbeam):
         distance(NX_NUMBER):
           unit: NX_LENGTH
+          exists: recommended
           doc: |
-            Distance of the point of evaluation of the beam from the sample surface.
+            The beam emitted by this source.
+            Should be named with the same appendix, e.g.,
+            for `source_probe` it should refer to `beam_probe`.
+            Refers to the same concept as /NXentry/NXinstrument/source_TYPE
+            and may be linked.
```

Comment:
Something must have gone wrong here. `beam_TYPE` does not have a docstring anymore and the docstring for `distance` is the same as the one in `NXsource/associated_beam`.
I suggest the following:
```yaml
beam_TYPE(NXbeam):
  doc: |
    Properties of the photon beam at a given location.
    Should be named with the same appendix as source_TYPE, e.g.,
    for `source_probe` it should refer to `beam_probe`.
  distance(NX_NUMBER):
    unit: NX_LENGTH
    exists: recommended
    doc: |
      Distance between the point where the current NXbeam instance is evaluating 
      the beam properties and the point where the beam interacts with the sample.
      For photoemission, the latter is the point where the the centre of the beam 
      touches the sample surface.
```
What do you think @rettigl @Tommaso-Pincelli?

Reply:
Looks good to me.

Reply:
Fixed, feel free to resolve